src/mixins/README.md

```diff
@@ -0,0 +1,128 @@
+
+It is also important for the constructor to call `super(args)` so `Application` continues to work as expected.
+```ts
+constructor(...args: any[]) {
+  super(args);
+}```
+
+### Binding via `ApplicationOptions`
+As mentioned earlier, since our `args` represents `ApplicationOptions`, we can make it possible for users to pass in their `Logger` implementations in a `loggers` array on `ApplicationOptions`. We can then read the array and automatically bind these for the user. 
+
+#### Example user experience
+```ts
+new LoggerMixin(Application)({
+  loggers: [ColorLogger],
+});
+```
+
+#### Example Implementation
+To implement this, we would check `this.options` to see if it has a `loggers` array and if so, bind it by calling the `.logger()` method. (More on that below).
+```ts
+if (this.options.loggers) {
+  for (const logger of this.options.loggers) {
+    this.logger(logger);
+  }
+}
+```
+
+### Binding via `.logger()`
+As mentioned earlier, we can add a new function to our `Application` class called `.logger()` into which a user would pass in their `Logger` implementation so we can bind it to the `loggers.*` key for them. We just add this new method on our partial Mixin class.
+```ts
+logger(logClass: Logger) {
+  const loggerKey = `loggers.${logClass.name}`;
+  this.bind(loggerKey).toClass(logClass);
+}
+```
+
+### Binding a `Logger` from a `Component`
+Our base class of `Application` already has a method that binds components. We can modify this method to continue binding a `Component` as usual but also binding any `Logger` instances provided by that `Component`. When modifying behavior of an existing method, we can ensure existing behavior by calling the `super.method()`. In our case the method is `.component()`.
+```ts
+component(component: Constructor<any>) {
+  super.component(component); // ensures existing behavior from Application
+  this.mountComponentLoggers(component);
+}
+```
+
+We have now modified `.component()` to do it's thing and then call our method `mountComponentLoggers()`. In this method is where we check for `Logger` implementations declared by the component in a `loggers` array by retrieving the instance of the `Component`. Then if `loggers` array exists, we bind the `Logger` instances as normal (by leveraging our `.logger()` method). 
+
+```ts
+mountComponentLoggers(component: Constructor<any>) {
+  const componentKey = `components.${component.name}`;
+  const compInstance = this.getSync(componentKey);
+
+  if (compInstance.loggers) {
+    for (const logger of compInstance.loggers) {
+      this.logger(logger);
+    }
+  }
+}
+```
+
+
+## Examples for using LoggerMixin
+```
+// Using the app's .logger() function. 
+class LoggingApplication extends LoggerMixin(Application) {
+  constructor(...args: any[]) {
+    super(...args);
+      const app = this;
+  }
+
+  app.logger(ColorLogger);
```

Comment:
This line should be moved outside of `class LoggingApplication`. The `app` variable is also not defined.

Did you perhaps want to call `this.logger(ColorLogger)` from the constructor?